lib/pwwka/persistent_transmitter.rb

```diff
@@ -0,0 +1,165 @@
+require_relative "publish_options"
+
+begin  # optional dependency
+  require 'resque'
+  require 'resque-retry'
+rescue LoadError
+end
+
+module Pwwka
+  # Primary interface for sending messages.
+  #
+  # Example:
+  #
+  #     # Send a message, blowing up if there's any problem
+  #     Pwwka::Transmitter.send_message!({ user_id: @user.id }, "users.user.activated")
+  #
+  #     # Send a message, logging if there's any problem
+  #     Pwwka::Transmitter.send_message_safely({ user_id: @user.id }, "users.user.activated")
+  class PersistentTransmitter
+
+    extend Pwwka::Logging
+    include Pwwka::Logging
+
+    DEFAULT_DELAY_BY_MS = 5000
+
+    attr_reader :channel_connector
+
+    def initialize
+      @channel_connector = ChannelConnector.new(connection_name: "p: #{Pwwka.configuration.app_id} #{Pwwka.configuration.process_name}".strip)
+    end
+
+    def self.connection(routing_key, type: nil, headers: nil, message_id: :auto_generate)
+      connection = self.new
+      yeild(connection)
+    ensure
+      connection.channel_connector.connection_close
```

Comment:
`connection` is really an instance of `PersistentTransmitter` so a name like `transmitter` would clarify what it is